api/proto/rpc/read_revisions.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package jj_api.rpc;
+
+import "objects/options.proto";
+import "objects/revset.proto";
+import "objects/revision.proto";
+
+message RevisionsRequest {
+  jj_api.objects.Options options = 1;
+
+  jj_api.objects.Revset revisions = 2;
+
+  int32 limit = 3;
+  // TODO: We have three options here:
+  // 1) Make this a string, and return the string
+  // 2) Make this similar to Revset, where we can construct it
+  // 3) Remove this from the API and put it in jj-cli entirely. Templates would,
+  //    instead of formatting a Commit object, format a Revision proto.
```

Comment:
To clarify, you're saying that you'd prefer that a `Template` should format a revision proto rather than the `Commit` object?

I'm a little confused because I had assumed that it was mutually exclusive with making templates a first-class citizen of the jj client. Could you explain what you're hoping for in more detail

My assumption was that either templates would format a Commit object inside the daemon, and be treated as a first-class citizen:
```
let revision = client.GetRevisions(ListRevisionsRequest(template="..."))
let rendered = template.rendered
```

Or templates would be purely for the use of `jj-cli`:
```
let revision = client.GetRevisions(ListRevisionsRequest());
let template = Template::new("...");
let rendered = template.render(revision);
```

And that in this second scenario, they would be unable to be a first-class citizen of the jj-client, since a python extension, for example, would be unable to use the rust-native implementation of `Template`

In the second scenario, we definitely aren't exposing the actual storage type, but in the first scenario, templates work  the same way they currently do, and I don't think that would involve exposing the actual storage type either (the end-user would just receive a string).

Reply:
> To clarify, you're saying that you'd prefer that a `Template` should format a revision proto rather than the `Commit` object?

I'm saying that the API server should have a distinct definition of what is stored, as not everyone will want a full commit. See #2988 for my thoughts on that, because I think Google wants to run `jj api` like the ClangD implementation. 

For the other part I'm gonna answer upthread.